mos-platform/atari2600-3e/link.ld

```diff
@@ -0,0 +1,86 @@
+/* TigerVision 3E ROM mapper */
+
+INCLUDE vcs.ld
+
+PROVIDE(__cart_rom_size = 32);
+PROVIDE(__cart_ram_size = 32);
+
+ASSERT(__cart_rom_size >= 6,
+       "ATARI 2600 3E: ROM must be at least 6 KiB")
+ASSERT(__cart_rom_size <= 32,
+       "ATARI 2600 3E: ROM must be no larger than 32 KiB")
+ASSERT(__cart_rom_size % 2 == 0,
+       "ATARI 2600 3E: ROM must be multiple of 2 KiB")
+ASSERT(__cart_ram_size <= 255,
+       "ATARI 2600 3E: RAM must be no larger than 255 KiB")
+ASSERT(__cart_ram_size % 2 == 0,
+       "ATARI 2600 3E: RAM must be multiple of 2 KiB")
+
+MEMORY {
+  rom0 (rx)     : ORIGIN = 0x1000, LENGTH = 0x800
+  rom1 (rx)     : ORIGIN = 0x3000, LENGTH = 0x800
+  rom2 (rx)     : ORIGIN = 0x5000, LENGTH = __cart_rom_size >= 8 ? 0x800 : 0
+  rom3 (rx)     : ORIGIN = 0x7000, LENGTH = __cart_rom_size >= 10 ? 0x800 : 0
+  rom4 (rx)     : ORIGIN = 0x9000, LENGTH = __cart_rom_size >= 12 ? 0x800 : 0
+  rom5 (rx)     : ORIGIN = 0xB000, LENGTH = __cart_rom_size >= 14 ? 0x800 : 0
+  rom6 (rx)     : ORIGIN = 0xD000, LENGTH = __cart_rom_size >= 16 ? 0x800 : 0
+  rom7 (rx)     : ORIGIN = 0xF000, LENGTH = __cart_rom_size >= 18 ? 0x800 : 0
+  rom8 (rx)     : ORIGIN = 0x1000, LENGTH = __cart_rom_size >= 20 ? 0x800 : 0
```

Comment:
I don't think this will work right. ORIGIN addresses should be unique per memory section, if they are to point to different banks. The traditional scheme for LLVM-MOS mappers is to use the bottom 16 bits to match the VMA (virtual memory address) of the 6502, and the top 16 bits to designate the bank used.

Reply:
I was wondering about that. Can I just make the top byte equal to the bank index, for example:

rom0 = 0x00003000
rom1 = 0x01003000
...etc

And for the fixed bank something like:

perm = 0xff00f800